RFCs/005-Framework-Extensibility-Custom-DataSource.md

```diff
@@ -0,0 +1,59 @@
+# RFC 005- Framework Extensibility for Custom DataSource
+
+## Summary
+This details the MSTest V2 framework extensibility for specifying custom data source for data driven tests.
+
+## Motivation
+Often times, custom data source is required for data driven tests. If the test framework provides an extensibility, custom data source functionality can be
+* Easily extended to use test data from custom data sources.
+
+## Detailed Design
+
+### Requirements
+1. Test data from custom data source should be easily associated with multiple test cases.
+
+### Proposed solution
+Here is a solution using which test data can be easily assocaited with test case.
+
+The test framework should define an interface `IDataSource` which can be extended to get data from custom data source. The defintion of this interface is as below :
+```
+    public interface IDataSource
+    {
+        /// <summary>
+        /// Gets the data from custom data source.
+        /// </summary>
+        public IEnumerable<object[]> GetData();
```

Comment:
Can you change this to `public IEnumerable<object[]> GetData(MethodInfo testMethod);` so that third party attributes can inspect the method before providing data. xUnit provides this and it's very handy.